src/creature.cpp

```diff
@@ -739,25 +743,27 @@ void Creature::setAttackedCreature(Creature* creature)
 	}
 
 	attackedCreature = creature;
-	creature->addFollower(this);
+	creature->addFollower(getCreature());
 	onAttackedCreature(attackedCreature);
 	attackedCreature->onAttacked();
 
-	for (Creature* summon : summons) {
-		summon->setAttackedCreature(creature);
+	for (const auto& summon : summons) {
+		if (summon) {
+			summon->setAttackedCreature(creature);
+		}
```

Comment:
Is it possible for the `pointer` to be `null` at this point?

Reply:
No, this is AI slop